### PR TITLE
add --createRandomTest option

### PR DESCRIPTION
**Pull request by winsvega**
remove legacy code

Usage:

``` ./testeth -t StateTestsGeneral -- --createRandomTest ```

### CODE REVIEW

File: test/tools/jsontests/StateTests.cpp [hunk 1]
```diff
@@ -49,7 +49,7 @@ void doStateTests(json_spirit::mValue& _v, bool _fillin)
 		string testname = i.first;
 		json_spirit::mObject& o = i.second.get_obj();
 
-		if (_fillin)
+		if (_fillin && !Options::get().createRandomTest)
```
Comment (pirapira) [inline test/tools/jsontests/StateTests.cpp]: What about implementing `TestOutputHelper::fillerFileName()`, which should be empty when there are no fillers?

File: test/tools/libtesteth/Options.cpp [hunk 2]
```diff
@@ -272,13 +272,28 @@ Options::Options(int argc, char** argv)
...
 		}
 	}
 
+	//check restrickted options
+	if (createRandomTest)
+	{
+		string optionlist = "trValueIndex, trGasIndex, trDataIndex, nonetwork, singleTest, performance, quadratic, memory, inputLimits, bigData, wallet, stats, filltests, fillchain";
+		if (trValueIndex >= 0 || trGasIndex >= 0 || trDataIndex >= 0 || nonetwork || singleTest
+			|| 	performance || quadratic || memory || inputLimits || bigData || wallet
+			|| 	stats || filltests || fillchain)
```
Comment (pirapira) [inline test/tools/libtesteth/Options.cpp]: The `<tab>` character should not be after `||`s.

File: test/tools/libtesteth/Options.cpp [hunk 3]
```diff
@@ -272,13 +272,28 @@ Options::Options(int argc, char** argv)
...
+		{
+			cerr << "--createRandomTest could not be used with one of the options: " + optionlist << endl;
```
Comment (pirapira) [inline test/tools/libtesteth/Options.cpp]: `cannot be used` is clearer to me.

File: test/tools/fuzzTesting/createRandomTest.cpp [hunk 4]
```diff
@@ -148,12 +94,13 @@ int checkRandomTest(std::function<void(json_spirit::mValue&, bool)> _doTests, js
 	return ret;
 }
 
-void fillRandomTest(std::function<void(json_spirit::mValue&, bool)> _doTests, std::string const& _testString, bool _debug)
+void fillRandomTest(std::function<void(json_spirit::mValue&, bool)> _doTests, std::string const& _testString)
 {
+	bool debug = (dev::test::Options::get().logVerbosity != dev::test::Verbosity::NiceReport);
```
Comment (pirapira) [inline test/tools/fuzzTesting/createRandomTest.cpp]: Usually this is `<=` or `>=`, but not `!=`.

File: test/tools/libtesteth/TestHelper.h [hunk 5]
```diff
@@ -166,7 +168,7 @@ json_spirit::mObject fillJsonWithState(eth::State const& _state, eth::AccountMas
 json_spirit::mObject fillJsonWithTransaction(eth::Transaction const& _txn);
 
 //Fill Test Functions
-int createRandomTest(std::vector<char*> const& _parameters);
+int createRandomTest();
```
Comment (pirapira) [inline test/tools/libtesteth/TestHelper.h]: Please document the return value.

File: test/tools/fuzzTesting/createRandomTest.cpp [hunk 6]
```diff
@@ -168,13 +115,21 @@ void fillRandomTest(std::function<void(json_spirit::mValue&, bool)> _doTests, st
...
 	}
+	catch (dev::Exception const& _e)
+	{
+		std::cerr << "Test fill exception: " << diagnostic_information(_e) << std::endl;
+	}
+	catch (std::exception const& _e)
+	{
+		std::cerr << "Test fill exception: " << _e.what() << std::endl;
+	}
 	catch(...)
```
Comment (chfast) [inline test/tools/fuzzTesting/createRandomTest.cpp]: This is too defensive. Do not catch exceptions that are not exceptions.

File: test/tools/fuzzTesting/createRandomTest.cpp [hunk 7]
```diff
@@ -168,13 +115,21 @@ void fillRandomTest(std::function<void(json_spirit::mValue&, bool)> _doTests, st
...
 	{
-		std::cerr << "Test fill exception!";
+		std::cerr << "Test fill exception!" << std::endl;
```
Comment (chfast) [inline test/tools/fuzzTesting/createRandomTest.cpp]: Do no use `std::endl`. Just use `\n`.
https://github.com/isocpp/CppCoreGuidelines/issues/357

File: test/tools/libtesteth/TestOutputHelper.cpp [hunk 8]
```diff
@@ -44,21 +44,15 @@ void TestOutputHelper::initTest(int _maxTests)
...
 	m_timer.restart();
 	m_currentTestCaseName = boost::unit_test::framework::current_test_case().p_name;
-	std::cout << "Test Case \"" + m_currentTestCaseName + "\": " << std::endl;
+	if (!Options::get().createRandomTest)
+		std::cout << "Test Case \"" + m_currentTestCaseName + "\": " << std::endl;
 	m_maxTests = _maxTests;
 	m_currTest = 0;
 }
 
 void TestOutputHelper::initTest(json_spirit::mValue& _v)
```
Comment (chfast) [inline test/tools/libtesteth/TestOutputHelper.cpp]: Why is this overload needed?

File: test/tools/libtesteth/TestOutputHelper.cpp [hunk 9]
```diff
@@ -32,10 +32,10 @@ using namespace boost;
...
 Timer TestOutputHelper::m_timer;
 size_t TestOutputHelper::m_currTest = 0;
-size_t TestOutputHelper::m_maxTests = 0;
+size_t TestOutputHelper::m_maxTests = 1;
 string TestOutputHelper::m_currentTestName = "n/a";
 string TestOutputHelper::m_currentTestCaseName = "n/a";
-string TestOutputHelper::m_currentTestFileName = "n/a";
+string TestOutputHelper::m_currentTestFileName;
 std::vector<TestOutputHelper::execTimeName> TestOutputHelper::m_execTimeResults;
 void TestOutputHelper::initTest(int _maxTests)
```
Comment (chfast) [inline test/tools/libtesteth/TestOutputHelper.cpp]: Use `size_t` type for argument if you want to pass results of `size()` functions.

File: test/tools/libtesteth/boostTest.cpp [hunk 10]
```diff
@@ -67,7 +67,7 @@ void createRandomTest()
 	//For no reason BOOST tend to remove valuable arg -t "TestSuiteName"
 	//And can't hadle large input stream of data
 	//so the actual test suite and raw test input is read into Options
-	if (dev::test::createRandomTest(parameters))
+	if (dev::test::createRandomTest())
 		throw framework::internal_error("Create Random Test Error!");
 	else
```
Comment (chfast) [inline test/tools/libtesteth/boostTest.cpp]: No `else` after throw.

File: test/tools/fuzzTesting/createRandomTest.cpp [hunk 11]
```diff
@@ -36,76 +36,22 @@ extern std::string const c_testExampleBlockchainTest;
...
-		else
-		if (arg == "--filldebug")
-			filldebug = true;
-	}
-
-	if (testSuite == "")
+	if (options.rCurrentTestSuite != test::c_StateTestsGeneral)
 	{
-		std::cerr << "Error! Test suite not supported! (Usage -t TestSuite)" << std::endl;
+		std::cerr << "Error! Test suite '" + options.rCurrentTestSuite + "' not supported! (Usage -t <TestSuite>)" << std::endl;
```
Comment (chfast) [inline test/tools/fuzzTesting/createRandomTest.cpp]: Don't use `std::endl`.

File: test/tools/libtesteth/Options.cpp [hunk 12]
```diff
@@ -272,13 +272,29 @@ Options::Options(int argc, char** argv)
...
 	}
 
+	//check restrickted options
+	if (createRandomTest)
+	{
+		if (trValueIndex >= 0 || trGasIndex >= 0 || trDataIndex >= 0 || nonetwork || singleTest
+			|| performance || quadratic || memory || inputLimits || bigData || wallet
+			|| stats || filltests || fillchain)
+		{
+			cerr << "--createRandomTest cannot be used with any of the options: " <<
```
Comment (chfast) [inline test/tools/libtesteth/Options.cpp]: Do you use `std::` prefix or not?

File: test/tools/fuzzTesting/createRandomTest.cpp (hunk 11 above)
Comment (winsvega) [inline test/tools/fuzzTesting/createRandomTest.cpp]: could this go to another PR?

File: test/tools/libtesteth/TestOutputHelper.cpp (hunk 8 above)
Comment (winsvega) [inline test/tools/libtesteth/TestOutputHelper.cpp]: well it could be removed. in another PR

File: test/tools/libtesteth/Options.cpp (hunk 12 above)
Comment (winsvega) [inline test/tools/libtesteth/Options.cpp]: in this file it is not used for some reason

File: test/tools/fuzzTesting/createRandomTest.cpp (hunk 11 above)
Comment (pirapira) [inline test/tools/fuzzTesting/createRandomTest.cpp]: Should this be a `\n` in the string literal?

Comment (chfast) [inline test/tools/fuzzTesting/createRandomTest.cpp]: Yes.

File: test/tools/libtesteth/TestOutputHelper.cpp (hunk 8 above)
Comment (pirapira) [inline test/tools/libtesteth/TestOutputHelper.cpp]: I'm on it.

Comment (pirapira) [inline test/tools/libtesteth/TestOutputHelper.cpp]: PR #4351

File: test/tools/fuzzTesting/createRandomTest.cpp (hunk 11 above)
Comment (pirapira) [inline test/tools/fuzzTesting/createRandomTest.cpp]: PR https://github.com/ethereum/cpp-ethereum/pull/4352

File: test/tools/libtesteth/TestOutputHelper.cpp (hunk 8 above)
Comment (winsvega) [inline test/tools/libtesteth/TestOutputHelper.cpp]: wait I have fixed that

File: test/tools/fuzzTesting/createRandomTest.cpp (hunk 11 above)
Comment (winsvega) [inline test/tools/fuzzTesting/createRandomTest.cpp]: https://github.com/ethereum/cpp-ethereum/pull/4340/commits/1845789686c68f19762d4d4391bc046520ab6fe3

File: test/tools/libtesteth/TestOutputHelper.cpp (hunk 8 above)
Comment (winsvega) [inline test/tools/libtesteth/TestOutputHelper.cpp]: https://github.com/ethereum/cpp-ethereum/pull/4340/commits/1845789686c68f19762d4d4391bc046520ab6fe3

File: test/tools/fuzzTesting/createRandomTest.cpp (hunk 11 above)
Comment (axic) [inline test/tools/fuzzTesting/createRandomTest.cpp]: @chfast why not use `std::endl`?

Comment (chfast) [inline test/tools/fuzzTesting/createRandomTest.cpp]: No. The `\n` is just better. The `std::endl` has a hidden flush included so decreases performance of logs.
https://github.com/isocpp/CppCoreGuidelines/issues/357
http://en.cppreference.com/w/cpp/io/manip/endl

Comment (axic) [inline test/tools/fuzzTesting/createRandomTest.cpp]: That's a weird design choice from C++.
